build/media_source/system/js/inline-editing.es6.js

```diff
@@ -0,0 +1,312 @@
+
+  const appendLoader = (element, hidden = true) => {
+    element.appendChild(loader);
+    element.appendChild(cover);
+    if (hidden) {
+      hideLoader();
+    } else {
+      showLoader();
+    }
+  };
+
+  container.appendChild(buttons);
+  appendLoader(container);
+
+  const format = (fieldHtml) => {
+    const parser = new DOMParser();
+    const doc = parser.parseFromString(fieldHtml, 'text/html');
+    const updatedField = doc.querySelector('.controls').children[0];
+
+    switch (inputField.tagName) {
+      case 'SELECT': {
+        const selectedOptions = updatedField.querySelectorAll('option:checked');
+        const display = [];
+
+        [].forEach.call(selectedOptions, (el) => {
+          display.push(el.text);
+        });
+
+        return display.join(', ');
+      }
+      case 'FIELDSET': {
+        const checkedOptions = updatedField.querySelectorAll('input:checked');
+        const display = [];
+
+        [].forEach.call(checkedOptions, (el) => {
+          display.push(el.labels[0].innerText.trim());
+        });
+
+        return display.join(', ');
+      }
+      default:
+        return updatedField.value;
+    }
+  };
+
+  form.addEventListener('submit', (event) => {
+    event.preventDefault();
+    if (selectedElement === null) {
+      return;
+    }
+
+    appendLoader(container, false);
+    const element = selectedElement;
+    const key = element.classList[1];
+    [inputField] = form.querySelector('.controls').children;
+    const data = options[key];
+
+    // Prepare the form data
+    const formData = new FormData(form);
+    formData.append('field_name', data.fieldName);
+    if (data.fieldGroup != null) {
+      formData.append('field_group', data.fieldGroup);
+    }
+    formData.append('task', `${data.controller}.saveInline`);
+    formData.append(Joomla.getOptions('csrf.token'), '1');
+
+    if (document.formvalidator.isValid(form) === false) {
+      return;
+    }
+
+    inputField.disabled = true;
+
+    // Make the ajax call
+    Joomla.request({
+      url: data.url,
+      method: 'POST',
+      data: formData,
+      onSuccess: (resp) => {
+        let response;
+
+        try {
+          response = JSON.parse(resp);
+        } catch (e) {
+          Joomla.renderMessages({ error: ['Invalid response.'] });
+          return;
+        }
+
+        if (response.success === true) {
+          const value = format(response.data.html);
+
+          element.innerHTML = value;
+          if (value === null || value === '') {
+            // remove the whole field from the dom.
+          }
+
+          element.classList.remove('d-none');
+          selectedElement = null;
+          previousValue = null;
+          document.documentElement.style.setProperty('--inline-editable-bg', 'blanchedalmond');
+          hideContainer();
+        } else {
+          if (response.message) {
+            Joomla.renderMessages({ error: [response.message] });
+          }
+          if (response.messages) {
```

Comment:
Something is wrong here, you are checking the same thing twice.

Reply:
It is different. First one is `message`, it is `messages` with a `s`.